spyder/plugins/editor/widgets/tests/test_codeeditor.py

```diff
@@ -366,5 +366,144 @@ def test_brace_match(editorbot):
+        expectedCol -= 1
+        assert editor.textCursor().columnNumber() == expectedCol
+        qtbot.keyPress(editor, Qt.Key_Up)
+        assert editor.textCursor().columnNumber() == expectedCol
+
+
+def test_editor_backspace_selection(editorbot):
+    """Regression test for issue spyder-ide/spyder#12663."""
+    qtbot, editor = editorbot
+    text = "0123456789\nabcdefghij\n9876543210\njihgfedcba\n"
+    editor.set_text(text)
+    expectedCol = 5
+    cursor = editor.textCursor()
+    cursor.setPosition(expectedCol)
+    editor.setTextCursor(cursor)
+
+    # This first subtest does not trigger the original bug
+    for press in range(3):
+        qtbot.keyPress(editor, Qt.Key_Left, Qt.ShiftModifier)
+    expectedCol -= 3
+    qtbot.keyPress(editor, Qt.Key_Backspace)
+    assert editor.textCursor().columnNumber() == expectedCol
+    qtbot.keyPress(editor, Qt.Key_Down)
+    assert editor.textCursor().columnNumber() == expectedCol
+
+    # However, this second subtest does trigger the original bug
+    for press in range(3):
+        qtbot.keyPress(editor, Qt.Key_Right, Qt.ShiftModifier)
+    qtbot.keyPress(editor, Qt.Key_Backspace)
+    assert editor.textCursor().columnNumber() == expectedCol
+    qtbot.keyPress(editor, Qt.Key_Down)
+    assert editor.textCursor().columnNumber() == expectedCol
+
+
+def test_editor_delete_char(editorbot):
+    """Regression test for issue spyder-ide/spyder#12663."""
+    qtbot, editor = editorbot
+    text = "0123456789\nabcdefghij\n9876543210\njihgfedcba\n"
+    editor.set_text(text)
+    expectedCol = 2
+    cursor = editor.textCursor()
+    cursor.setPosition(expectedCol)
+    editor.setTextCursor(cursor)
+    for line in range(3):
+        qtbot.keyPress(editor, Qt.Key_Delete)
+        assert editor.textCursor().columnNumber() == expectedCol
+        qtbot.keyPress(editor, Qt.Key_Down)
+        assert editor.textCursor().columnNumber() == expectedCol
+
+    for line in range(3):
+        qtbot.keyPress(editor, Qt.Key_Delete)
+        assert editor.textCursor().columnNumber() == expectedCol
+        qtbot.keyPress(editor, Qt.Key_Up)
+        assert editor.textCursor().columnNumber() == expectedCol
+
+
+def test_editor_delete_selection(editorbot):
+    """Regression test for issue spyder-ide/spyder#12663."""
+    qtbot, editor = editorbot
+    text = "0123456789\nabcdefghij\n9876543210\njihgfedcba\n"
+    editor.set_text(text)
+    expectedCol = 5
+    cursor = editor.textCursor()
+    cursor.setPosition(expectedCol)
+    editor.setTextCursor(cursor)
+
+    # This first subtest does not trigger the original bug
+    for press in range(3):
+        qtbot.keyPress(editor, Qt.Key_Left, Qt.ShiftModifier)
+    expectedCol -= 3
+    qtbot.keyPress(editor, Qt.Key_Delete)
+    assert editor.textCursor().columnNumber() == expectedCol
+    qtbot.keyPress(editor, Qt.Key_Down)
+    assert editor.textCursor().columnNumber() == expectedCol
+
+    # However, this second subtest does trigger the original bug
```

Comment:
This means that this part will catch a regression in case we change things and they don't stop working as they are in this PR?

Reply:
The second subtest (last two asserts) would catch a regression, yes. In the sense that the test will fail in 4.2.1, but succeds with things as they are in this PR.